tests/components/lock/test_init.py

```diff
@@ -2,328 +2,261 @@
+        data[ATTR_CODE] = code
 
+    await hass.services.async_call(DOMAIN, service, data, blocking=True)
 
-async def test_lock_states(hass: HomeAssistant) -> None:
-    """Test lock entity states."""
 
-    lock = MockLockEntity()
-    lock.hass = hass
+async def test_lock_default(hass: HomeAssistant, mock_lock: MockLock) -> None:
+    """Test lock entity with defaults."""
 
-    assert lock.state is None
+    assert mock_lock.code_format is None
+    assert mock_lock.state is None
 
-    lock._attr_is_locking = True
-    assert lock.is_locking
-    assert lock.state == STATE_LOCKING
 
-    await lock.async_handle_lock_service()
-    assert lock.is_locked
-    assert lock.state == STATE_LOCKED
+async def test_lock_states(hass: HomeAssistant, mock_lock: MockLock) -> None:
+    """Test lock entity states."""
 
-    lock._attr_is_unlocking = True
-    assert lock.is_unlocking
-    assert lock.state == STATE_UNLOCKING
+    assert mock_lock.state is None
 
-    await lock.async_handle_unlock_service()
-    assert not lock.is_locked
-    assert lock.state == STATE_UNLOCKED
+    mock_lock._attr_is_locking = True
+    assert mock_lock.is_locking
+    assert mock_lock.state == STATE_LOCKING
 
-    lock._attr_is_jammed = True
-    assert lock.is_jammed
-    assert lock.state == STATE_JAMMED
-    assert not lock.is_locked
+    await help_test_async_lock_service(hass, SERVICE_LOCK)
+    assert mock_lock.is_locked
+    assert mock_lock.state == STATE_LOCKED
 
+    mock_lock._attr_is_unlocking = True
+    assert mock_lock.is_unlocking
+    assert mock_lock.state == STATE_UNLOCKING
 
-async def test_set_default_code_option(
-    hass: HomeAssistant,
-    entity_registry: er.EntityRegistry,
-    enable_custom_integrations: None,
-) -> None:
-    """Test default code stored in the registry."""
-
-    entry = entity_registry.async_get_or_create("lock", "test", "very_unique")
-    await hass.async_block_till_done()
+    await help_test_async_lock_service(hass, SERVICE_UNLOCK)
+    assert not mock_lock.is_locked
+    assert mock_lock.state == STATE_UNLOCKED
 
-    platform = getattr(hass.components, "test.lock")
-    platform.init(empty=True)
-    platform.ENTITIES["lock1"] = platform.MockLock(
-        name="Test",
-        code_format=r"^\d{4}$",
-        supported_features=LockEntityFeature.OPEN,
-        unique_id="very_unique",
-    )
+    mock_lock._attr_is_jammed = True
+    assert mock_lock.is_jammed
+    assert mock_lock.state == STATE_JAMMED
+    assert not mock_lock.is_locked
 
-    assert await async_setup_component(hass, "lock", {"lock": {"platform": "test"}})
-    await hass.async_block_till_done()
 
-    entity0: MockLock = platform.ENTITIES["lock1"]
-    entity_registry.async_update_entity_options(
-        entry.entity_id, "lock", {CONF_DEFAULT_CODE: "1234"}
-    )
-    await hass.async_block_till_done()
+@pytest.mark.parametrize(
+    ("default_code", "code_format", "supported_features"),
+    [("1234", r"^\d{4}$", LockEntityFeature.OPEN)],
+)
```

Comment:
I don't know if this is a usual way tests and fixtures are made/used in HA, however IMHO, using `parametrize` like this makes it look a little bit magic and implicit: one has to know that the parameters will be used in the fixture named `mock_lock`, which is in fact the `setup_lock_platform_test_entity` fixture, which is in another file (`conftest.py`). And using only one line of parameters with `parametrize` feels like a diverted use of the functionality.

Also I don't know how well IDEs would deal with this and I think that other coders/newcomers would find it hard to read and understand.

I would personally prefer to be explicit, and maybe use a standard function rather than a fixture with a distant `parametrize`. I would call that function at the first line with the relevant `default_code`, `code_format` and `supported_features` values.

This comment applies to the other tests.

my 2 p

Reply:
This is a best practice on parametrizing fixtures. There are serveral other examples in `tests/contest.py` that make use of this.

Reply:
Thanks. I will have a deeper look.